### PR TITLE
std: freebsd build update.

### PR DESCRIPTION
**Pull request by devnexen**
since freebsd 11 had been removed, minimum is now 12.

### CODE REVIEW

**Comment by cuviper** [comment]
> since freebsd 11 had been removed

Has it? #89058 is still open...

**Comment by workingjubilee** [comment]
It was accepted however, no?

File: library/std/build.rs [hunk 1]
```diff
@@ -4,10 +4,8 @@ fn main() {
     println!("cargo:rerun-if-changed=build.rs");
     let target = env::var("TARGET").expect("TARGET was not set");
     if target.contains("freebsd") {
-        if env::var("RUST_STD_FREEBSD_12_ABI").is_ok() {
-            println!("cargo:rustc-cfg=freebsd12");
-        } else if env::var("RUST_STD_FREEBSD_13_ABI").is_ok() {
-            println!("cargo:rustc-cfg=freebsd12");
+        println!("cargo:rustc-cfg=freebsd12");
```
Comment (cuviper) [inline library/std/build.rs:7]: With this minimum, we should drop the cfg altogether and remove `cfg(not(freebsd12))` code.

Comment (cuviper) [reply library/std/build.rs:7]: AFAICS just a couple functions in `library/std/src/os/freebsd/fs.rs` are using that.

File: library/std/build.rs [hunk 2]
```diff
@@ -3,14 +3,7 @@ use std::env;
...
     let target = env::var("TARGET").expect("TARGET was not set");
-    if target.contains("freebsd") {
-        if env::var("RUST_STD_FREEBSD_12_ABI").is_ok() {
-            println!("cargo:rustc-cfg=freebsd12");
-        } else if env::var("RUST_STD_FREEBSD_13_ABI").is_ok() {
-            println!("cargo:rustc-cfg=freebsd12");
-            println!("cargo:rustc-cfg=freebsd13");
-        }
-    } else if target.contains("linux")
+    if target.contains("linux")
```
Comment (cuviper) [inline library/std/build.rs:6]: I think we need freebsd to be in this big common list, otherwise it will fall into the `else restricted-std`.

**Comment by cuviper** [comment]
Thanks!

@bors r+ rollup

**Pull request merged**